<?xml version="1.0"?>
<launch>
  <arg name="camera_name" default="newheads4" />

  <!-- if show a image_view window subscribed to the generated stream -->
  	<arg name="visualize" default="false"/>
    <node name="$(arg camera_name)_headsvel_node" pkg="movehead" type="getheadsvel_v4.py" respawn="false" output="screen" required="true">
      <!-- max_head_age tells how many frames to follow a head before giving up -->
      <param name="max_head_age" value="30" />
    </node>
    <node name="$(arg camera_name)_im_pub" pkg="movehead" type="showheads2.py" respawn="false" output="screen" required="true">
      <param name="image_topic" value="/local/rgb/image_raw"/>

      <remap from="heads_im" to="heads_im_new"/>
      <remap from="heads" to="newheads"/>
    </node>
    <node if="$(arg visualize)" name="$(arg camera_name)_normal_image_view" pkg="image_view" type="image_view">
      <remap from="image" to="heads_im_new" />
    </node>

  <node name="$(arg camera_name)_flowheads_node" pkg="movehead" type="showheads2.py" respawn="false" output="screen" required="true">
    <param name="image_topic" value="/df_node/flow_xy"/>
    <remap from="heads_im" to="heads_im_flow"/>
    <remap from="heads" to="newheads_flow"/>
  </node>

  <node if="$(arg visualize)" name="$(arg camera_name)_flow_image_view" pkg="image_view" type="image_view">
    <remap from="image" to="heads_im_flow" />
  </node>
</launch>
